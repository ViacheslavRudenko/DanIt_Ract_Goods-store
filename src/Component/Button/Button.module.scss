.active {
  background-color: initial;
  border: 1px solid grey;
  border-radius: 2px;
  padding: 10px 30px;
  justify-content: center;
  border-radius: 5px;
}

.btn {
  text-align: center;
  justify-content: center;
  padding: 10px 30px;
  border: none;
  border-radius: 5px;
  color: #fff;
  font-size: larger;
  background-color: grey;
}
