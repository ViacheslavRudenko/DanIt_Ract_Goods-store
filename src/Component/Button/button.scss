.btn {
  &__item {
    margin: 0 auto;
    justify-content: center;
    padding: 10px 30px;
    border: none;
    border-radius: 5px;
    color: #fff;
    font-size: larger;
    background-color: grey;
  }
}
