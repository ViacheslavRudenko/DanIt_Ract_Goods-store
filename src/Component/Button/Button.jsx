import "./button.scss";
import PropTypes from "prop-types";
import styles from "./Button.module.scss";

export default function Button({ btn, actionWithModal }) {
  return (
    <>
      {btn.map((e) => (
        <button
          key={e.id}
          id={e.id}
          className={e.isActive ? styles.btn : styles.active}
          onClick={actionWithModal}
        >
          {e.text}
        </button>
      ))}
    </>
  );
}

Button.propTypes = {
  btn: PropTypes.array,
  setLS: PropTypes.func,
};
