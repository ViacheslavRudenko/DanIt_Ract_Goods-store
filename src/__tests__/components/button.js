import Button from "../../Component/Button/Button";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";

describe("<Button/> component", () => {
  const btn = [{ id: 1, text: "Add to cart" }];

  const onChange = jest.fn();
  const BtnComponent = <Button btn={btn} actionWithModal={onChange} />;

  it("renders correctly", () => {
    const tree = renderer.create(BtnComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render component", () => {
    render(BtnComponent);
    screen.getByRole("button");
  });

  it("should render text in component", () => {
    render(BtnComponent);
    screen.getByText(/add to cart/i);
  });

  it("should pass OnClick if btn is clicked", () => {
    render(BtnComponent);

    userEvent.pointer({
      keys: "[MouseLeft]",
      target: screen.getByRole("button"),
    });

    expect(onChange).toHaveBeenCalled();
  });
});
