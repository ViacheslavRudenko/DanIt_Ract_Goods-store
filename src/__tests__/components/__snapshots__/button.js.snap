// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> component renders correctly 1`] = `
<button
  className="active"
  id={1}
  onClick={[MockFunction]}
>
  Add to cart
</button>
`;
